src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -493,4 +511,56 @@ public static Object coerceBodyType(InputStream body, ResponseBodyType as,
 
         return response;
     }
+
+    private static Timer.Context startTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
+                    requestLine.getMethod());
+            return registry.timer(name).time();
+        } else {
+            return null;
+        }
+    }
+
+    private static Timer.Context startUnbufferedStreamTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
+                    requestLine.getMethod(), "unbuffered_stream");
+            return registry.timer(name).time();
+        } else {
+            return null;
+        }
+    }
+
+    public static SortedMap<String, Timer> getClientMetrics(MetricRegistry metricRegistry){
```

Comment:
nit: it'd be nice to keep all of the public methods in one spot and the private methods in another spot, but we've kind of already broken that pattern before your PR, so, maybe not worth trying to fix everything.

Would be good to have  javadocs on the public methods, though.